### PR TITLE
unsafe-block test case and detector

### PR DESCRIPTION
**Pull request by faculerena**

### CODE REVIEW

**Comment by faculerena** [comment]
review this after #50

File: test-cases/avoid-unsafe-block/avoid-unsafe-block-1/vulnerable-example/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+lto = true
+
+[profile.release-with-logs]
+inherits = "release"
+debug-assertions = true
+
+[workspace.metadata.dylint]
+libraries = [
+    { path = "/home/flerena/coinfabrik/soroban/scout-soroban/detectors/unsafe-block"},
+]
```
Comment (jgcrosta) [inline test-cases/avoid-unsafe-block/avoid-unsafe-block-1/vulnerable-example/Cargo.toml:35]: Remove

File: test-cases/avoid-unsafe-block/avoid-unsafe-block-1/vulnerable-example/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+soroban-sdk = { version = "20.0.0-rc2" }
+
+[dev_dependencies]
+soroban-sdk = { version = "20.0.0-rc2", features = ["testutils"] }
```
Comment (jgcrosta) [inline test-cases/avoid-unsafe-block/avoid-unsafe-block-1/vulnerable-example/Cargo.toml:13]: Version -> "20.0.0"

File: test-cases/avoid-unsafe-block/avoid-unsafe-block-1/remediated-example/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+soroban-sdk = { version = "20.0.0-rc2" }
+
+[dev_dependencies]
+soroban-sdk = { version = "20.0.0-rc2", features = ["testutils"] }
```
Comment (jgcrosta) [inline test-cases/avoid-unsafe-block/avoid-unsafe-block-1/remediated-example/Cargo.toml:13]: Version -> "20.0.0"

File: detectors/avoid-unsafe-block/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "avoid-unsafe-block"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+clippy_utils = { workspace = true }
```
Comment (jgcrosta) [inline detectors/avoid-unsafe-block/Cargo.toml:10]: This is not necessary

File: detectors/avoid-unsafe-block/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                if let ExprKind::Block(block, _) = expr.kind {
+                    if block.rules
+                        == rustc_hir::BlockCheckMode::UnsafeBlock(
+                            rustc_hir::UnsafeSource::UserProvided,
+                        )
+                    {
+                        self.unsafe_blocks.push(Some(expr.span));
+                    } else {
+                        self.unsafe_blocks.push(None);
+                    }
```
Comment (jgcrosta) [inline detectors/avoid-unsafe-block/src/lib.rs:81]: This is the same as not pushing anything

**Review by jgcrosta — APPROVED** [review]

**Pull request merged**